src/main/kotlin/org/rust/ide/inspections/RsSortImplTraitMembersInspection.kt

```diff
@@ -33,9 +33,11 @@ class RsSortImplTraitMembersInspection : RsLocalInspectionTool() {
 
     companion object {
         private fun sortedImplItems(implItems: List<RsItemElement>, traitItems: List<RsItemElement>): List<RsItemElement>? {
-            val implItemMap = implItems.associate { it.key() to it }
-            val sortedImplItems = traitItems.mapNotNull { implItemMap[it.key()] }
-            if (sortedImplItems.size != implItems.size || sortedImplItems == implItems) return null
+            val traitItemMap = traitItems.withIndex().associate { it.value.key() to it.index }
+            val implItemsInTrait = implItems.filter { traitItemMap.contains(it.key()) }
+            if (implItemsInTrait.size < implItems.size) return null
```

Comment:
Do I understand correctly that here you check that all items from `implItems` are in the corresponding trait item?
If it's true, looks like that you let's just check this case more explicitly (for instance, somehow like this`if (implItems.any { it.key() !in traitItemMap }) return null`)

Reply:
Oh, let's create `Testmark` for this case
The goal of `Testmark` is to link rare cases with their tests and ensure that you don't drop support of such cases during refactorings
You can see example of testmark using [here](https://github.com/intellij-rust/intellij-rust/blob/master/src/main/kotlin/org/rust/ide/inspections/import/AutoImportFix.kt#L78)
Don't forget to use it in test - just pass testmark object in `check*` function call in the corresponding test

Reply:
Done! 🙂